backend/src/auth_handlers.py

```diff
@@ -1,131 +1,63 @@
     :return: flask response object
     """
-    if request.get_json().get("user") is None:
-        return server.err_out(401, "user is a required field")
     # get user info from front end
     try:
-        user_data = request.get_json().get("user")
-        userd = User(user_data)
-    except KeyError as err:
-        return server.err_out(401, "'{}' is a required field".format(str(err)))
-
-    # check if user already exists
-    if server.db.username_already_exists(userd.username):
-        return server.err_out(
-            409, "username {} already exists".format(userd.username))
-
+        user_data = request.get_json() #dictionary of input
+        username = user_data.get('user_name')
+        password = user_data.get('password')
+        role = user_data.get('role')
+        is_admin = bool(user_data.get('is_admin'))
+        if username and password and role and is_admin:
+            created_user = app.services.user_service().add_user(username,password,role,is_admin)
+            return jsonify({'jwt': app.services.auth_service().encode_jwt(created_user)})
+        else:
+            return jsonify(code=400, error='Request is invalid'), 400
+    except KeyError:
+        return jsonify(code=400, error='Request is invalid'), 400
     # create new entry in the database
-    err = server.db.add(userd)
-    if err is not None:
-        return server.err_out(
-            500, "Could not add user to the database: {}".format(err))
-
-    return jsonify({
-        "user": userd.get_info(),
-        "jwt": encodeJWT(userd),
-    })
 
+    # TODO: confirm somehow??
 
-def rotateServerKeyIfNeeded():
-    """
-    rotates the server key if the expiration has passed
-    """
-    if server.tokenFutureRotationDate < datetime.now(timezone.utc):
-        logging.debug(
-            "rotating key, at least {} has passed since last rotation".format(
-                server.tokenRotationIntervalSec))
-        "{}-{}".format(secrets.token_hex(64), time.time())
-        server.tokenFutureRotationDate = datetime.now(
-            timezone.utc) + timedelta(seconds=server.tokenRotationIntervalSec)
+    # create a JWT token and return to the front end
 
+   
 
+@auth_module.route('/login', methods=['POST'])
 def login():
     """login an existing user
     :return: flask response object
     """
-    rotateServerKeyIfNeeded()
     # get user and password from front end
     u = request.get_json().get("username")
     p = request.get_json().get("password")
     # validate that user and password is valid
-    (user, err) = server.db.validate_login(u, p)
-    if err is not None:
-        return server.err_out(401, err)
-    # create a JWT token and return to front end
-    return jsonify({'jwt': encodeJWT(user)})
-
-
+    user = app.services.auth_service().validate_login(u, p)
+    if user:
+        # create a JWT token and return to front end
+        return jsonify({'jwt': app.services.auth_service().encode_jwt(user)})
+    else:
```

Comment:
```suggestion
```
don't need `else..return`, applies to other instances

Reply:
Done. I've been coding in Scala for a while where the use of return statements is generally discouraged. It still feels weird to use them :)